lib/entry.js

```diff
@@ -121,14 +135,18 @@ function SalesforceEntity(callScope) {
       }
     } while (hasMorePages);
 
-    snapshot.previousLastModified = lastSeenTime;
-    console.log('emitting new snapshot: %j', snapshot);
-    emitSnapshot(snapshot);
+    if (snapshot.previousLastModified !== lastSeenTime) {
+        snapshot.previousLastModified = lastSeenTime;
+        console.log('emitting new snapshot: %j', snapshot);
+        emitSnapshot(snapshot);
+    }
     emitEnd();
 
     function processResults(err, records) {
       if (err) {
-        onError(err);
+          onError(err);
+          hasMorePages = false;
```

Comment:
Why set this to false?

Reply:
For exiting  from while

Reply:
Can we do something like `throw createPresentableError(err);` instead?  I think that may be clearer.

Reply:
Changed